### PR TITLE
fix: Keep `sync_container_lifecycles()` bgtask alive in a loop.

### PR DESCRIPTION
**Pull request by fregataa**
### TL; DR
`sync_containers_lifecycle()` should work

### Detail
We have 3 different sources of Kernel / Container data
1. Manager side DB
2. Agent side `kernel_registry`
3. Agent side actual docker containers

I tried to synchronize ALL Agent's data to Manager's DB but synchronizing actual containers to `kernel_registry` or manager's data needs to process container destruction tasks which take a long time.
Since `sync_containers_lifecycle()` task synchronizes actual containers to `kernel_registry` periodically in background, resource-sync API only needs to sync `kernel_registry` to Manager side DB if it is ensured that the background task stays alive and works as we expect.

### Fix
- Keep `sync_container_lifecycles()` bgtask alive in a loop by handling any exceptions raised in the task.
- In the try-except context of container creation, **do not** use unbound local variable

### Enhance
- Fetch all containers eagerly in `sync_container_lifecycles()`

**Checklist:** (if applicable)

- [x] Milestone metadata specifying the target backport version

### CODE REVIEW

**Comment by fregataa** [comment]
* **#2178** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai/2178?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fregataa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>
